runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -3056,6 +3056,11 @@ ClientSessionData::cacheIProfilerInfo(TR_OpaqueMethodBlock *method, uint32_t byt
    auto it = j9methodMap.find((J9Method*)method);
    if (it != j9methodMap.end())
       {
+      // Check and update if method is compiled when collecting profiling data
+      bool isCompiled = TR::CompilationInfo::isCompiled((J9Method*)method);
```

Comment:
You could move this code down when the iProfilerMap is allocated. Because all the entries for a method are added more or less at the same time, you only need to set that flag once per j9method.

Reply:
If I move it down where it is allocated, I'm worried that the flag won't be ever set to true because the only time it updates is when it is created. Could we maybe add a `if debug` around this code so it doesn't run for prod version?